docs/src/guides/pip.md

```diff
@@ -0,0 +1,257 @@
+
+- The first one builds [Pillow](https://python-pillow.org/) from upstream sources fetched from PyPi.
+- The second one builds a fictional python project living in the same repository as the nix sources
+  and a development environment around it.
+
+## Start with a flake
+
+We start both examples by creating a new git repository and adding almost the same `flake.nix` template we already used in [Getting Started](./getting-started.md#start-a-project). The only difference
+are the packages name, `default` instead of `hello`:
+
+```nix title="flake.nix"
+{
+  inputs = {
+    dream2nix.url = "github:nix-community/dream2nix";
+    nixpkgs.follows = "dream2nix/nixpkgs";
+  };
+
+  outputs = {
+    self,
+    dream2nix,
+    nixpkgs,
+  }:
+  let
+      eachSystem = nixpkgs.lib.genAttrs [
+      "aarch64-darwin"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "x86_64-linux"
+    ];
+  in {
+  packages = eachSystem (system: {
+    default = dream2nix.lib.evalModules { # (1)
+      packageSets.nixpkgs = nixpkgs.legacyPackages.${system};
+      modules = [
+        ./default.nix (2)
+        {
+          paths.projectRoot = ./.;
+          paths.projectRootFile = "flake.nix";
+          paths.package = ./.;
+        }
+      ];
+    };
+  });
+}
+```
+
+1. We call our package attribute `default` here...
+2. ...and the nix file `default.nix` here.
+
+## Example: Pillow
+
+Things get a bit more interesting in `default.nix` where we define a package module which fetches Pillow from pypi and builds it with minial features - just JPEG support. Click the plus to expand any code annotation below for details.
+The code we are going to end up with is also available in [./examples/packages/languages/python-packaging-pillow](https://github.com/nix-community/dream2nix/tree/main/examples/packages/languages/python-packaging-pillow).
+
+### Code
+
+```nix title="default.nix"
+{
+  config,
+  lib,
+  dream2nix,
+  ...
+}: {
+  imports = [
+    dream2nix.modules.dream2nix.pip # (1)
+  ];
+
+  deps = {nixpkgs, ...}: {
+    python = nixpkgs.python311; # (2)
+    inherit # (3)
+      (nixpkgs)
+      pkg-config
+      zlib
+      libjpeg
+      ;
+  };
+
+  name = "pillow"; # (4)
+  version = "10.4.0";
+
+  mkDerivation = { # (5)
+    nativeBuildInputs = [
+      config.deps.pkg-config
+    ];
+    propagatedBuildInputs = [
+      config.deps.zlib
+      config.deps.libjpeg
+    ];
+  };
+
+  buildPythonPackage = { # (6)
+    pythonImportsCheck = [
+      "PIL"
+    ];
+  };
+
+  pip = {
+    requirementsList = ["${config.name}==${config.version}"]; # (7)
+    pipFlags = [ # (8)
+      "--no-binary"
+      ":all:"
+    ];
+  };
+}
+```
+
+1. Import the dream2nix [pip module](../reference/pip/index.md) into our module.
+2. Declare external dependencies, like the python interpreter to use and libraries from nixpkgs. We use whatever the latest `python3` in nixpkgs is as our python.
```

Comment:
The dream module is set specifically to python311, so this isn't true.

Reply:
Very good catch, thanks! Copied this a bit too quickly when changing examples.